Obok_plugin/obok/obok.py

```diff
@@ -376,36 +376,50 @@ def __init__ (self, serials = [], device_path = None, desktopkobodir = u""):
                 elif sys.platform.startswith('darwin'):
                     self.kobodir = os.path.join(os.environ['HOME'], "Library", "Application Support", "Kobo", "Kobo Desktop Edition")
                 elif sys.platform.startswith('linux'):
-
-                    #sets ~/.config/calibre as the location to store the kobodir location info file and creates this directory if necessary
-                    kobodir_cache_dir = os.path.join(os.environ['HOME'], ".config", "calibre")
+                    # Since on Linux, you have to run Kobo Desktop within Wine,
+                    # there is no guarantee where Kobo.sqlite (and the rest of
+                    # the kobo directory) might be.
+                    #
+                    # It should also be noted that Kobo Desktop will store all
+                    # of it files in the current directory where you run it,
+                    # meaning that a user might accidentally create several
+                    # Kobo.sqlite files which are all separate and then be
+                    # confused why Obok can't find any of the new books. Sadly
+                    # there isn't a trivial way to deal with this.
+
+                    # We cache the kobodir we find in ~/.config/calibre.
+                    kobodir_cache_dir = os.path.join(os.environ['HOME'], ".config", "calibre", "plugins", "obok")
                     if not os.path.isdir(kobodir_cache_dir):
                         os.mkdir(kobodir_cache_dir)
-                    
-                    #appends the name of the file we're storing the kobodir location info to the above path
-                    kobodir_cache_file = str(kobodir_cache_dir) + "/" + "kobo location"
-                    
-                    """if the above file does not exist, recursively searches from the root
-                    of the filesystem until kobodir is found and stores the location of kobodir
-                    in that file so this loop can be skipped in the future"""
-                    original_stdout = sys.stdout
-                    if not os.path.isfile(kobodir_cache_file):
-                        for root, dirs, files in os.walk('/'):
-                            for file in files:
-                                if file == 'Kobo.sqlite':
-                                    kobo_linux_path = str(root)
-                                    with open(kobodir_cache_file, 'w') as f:
-                                        sys.stdout = f
-                                        print(kobo_linux_path, end='')
-                                        sys.stdout = original_stdout
-
-                    f = open(kobodir_cache_file, 'r' )
-                    self.kobodir = f.read()
-
-            # desktop versions use Kobo.sqlite
+                    kobodir_cache_file = os.path.join(kobodir_cache_dir, "kobo-location")
```

Comment:
I moved this into `~/.config/calibre/plugins/obok` since that better matches the internal calibre configuration paths than just dumping it in `~/.config/calibre`. Users upgrading shouldn't notice a huge difference (thanks to the `$HOME` optimisation, the first search will be fairly quick) though in theory (since there might be multiple `Kobo.sqlite` files on their machine) this might change which Kobo data directory is used.